substrate/client/network/transactions/src/config.rs

```diff
@@ -57,13 +57,13 @@ pub type TransactionImportFuture = Pin<Box<dyn Future<Output = TransactionImport
 /// Transaction pool interface
 pub trait TransactionPool<H: ExHashT, B: BlockT>: Send + Sync {
 	/// Get transactions from the pool that are ready to be propagated.
-	fn transactions(&self) -> Vec<(H, B::Extrinsic)>;
+	fn transactions(&self) -> Vec<(H, Arc<B::Extrinsic>)>;
 	/// Get hash of transaction.
-	fn hash_of(&self, transaction: &B::Extrinsic) -> H;
+	fn hash_of(&self, transaction: &Arc<B::Extrinsic>) -> H;
 	/// Import a transaction into the pool.
 	///
 	/// This will return future.
-	fn import(&self, transaction: B::Extrinsic) -> TransactionImportFuture;
+	fn import(&self, transaction: Arc<B::Extrinsic>) -> TransactionImportFuture;
```

Comment:
This currently does not make sense. We would need to change [`submit_one`](https://github.com/paritytech/polkadot-sdk/blob/568231a9a85d94954c002532a0f4351a3bb59e83/substrate/client/transaction-pool/api/src/lib.rs#L263)/ [`submit_and_watch`](https://github.com/paritytech/polkadot-sdk/blob/568231a9a85d94954c002532a0f4351a3bb59e83/substrate/client/transaction-pool/api/src/lib.rs#L272) API to accept ARC.
Otherwise we create arc, then dereference it and create new one withing transcaction pool implementation.